metrics/perplexity/perplexity.py

```diff
@@ -188,3 +185,46 @@ def _compute(self, predictions, model_id, batch_size: int = 16, add_start_token:
             ppls += perplexity_batch.tolist()
 
         return {"perplexities": ppls, "mean_perplexity": np.mean(ppls)}
+
+    def _compute(self, logits=None, labels=None, attention_mask=None):
+        """
+        Computes perplexity according to a set of logits, labels, and an attention mask.
+        Args:
+            logits (`ndarray`): Tensor-like, of shape [batch size, sequence length, vocab size]
+            labels (`ndarray`): Tensor-like, of shape [batch, sequence length]
+            attention_mask (`ndarray`): Tensor-like, of shape [batch, sequence length]
+
+        Returns:
+            (`dict`): Dictionary containing perplexity for each example and mean perplexity.
+        """
+        ppls = torch.exp(
+            (CrossEntropyLoss(reduction="none")(logits.transpose(1, 2), labels) * attention_mask).sum(1)
+            / attention_mask.sum(1)
+        ).tolist()
+        return {"perplexities": ppls, "mean_perplexity": np.mean(ppls)}
+
+    def compute_perplexity_with_pretrained_model(
```

Comment:
Since this is a breaking change I don't think we actually need that option.

Reply:
also this is essentially what the `measurement` version does so a user could just use that.

Reply:
Sure thing, can remove — in that case, do you want any of the changes to `compute` reflected on the measurement version of perplexity at all? I was intending to reflect them so the APIs remain mirrored for the case where someone tries to use the measurement/metric versions of ppl interchangably.